modin/core/execution/ray/common/engine_wrapper.py

```diff
@@ -307,6 +307,32 @@ def post_materialize(self, materialized):
         """
         raise NotImplementedError()
 
+    def __reduce__(self):
+        """
+        Replace this hook with the materialized object on serialization.
+
+        Returns
+        -------
+        tuple
+        """
+        data = RayWrapper.materialize(self)
+        return int if isinstance(data, int) else self._get, (data,)
```

Comment:
I was also confused by this line, let's make it more obvious
```suggestion
        reconstructor = int if isinstance(data, int) else self._get
        return reconstructor, (data,)
```

Reply:
actually, can it simply be?
```python
def __reduce__(self):
    return (lambda x: x), (data,)
```

Reply:
I use `len(pickle.dumps(hook))` to calculate the length of the serialized data. In case of `int` this is 37, `self._get` - 96, ` lambda x: x` - `Can't pickle local object`. Anyway, creating and serializing a new object each time when we need to serialize a single int - this is a redundant overhead. I'm not familiar with the ray serializer implementation, but, if it's smart enough, it should not serialize the same object multiple times. For example, when serializing 100 hook objects, in case of a static function, this function + 100 tuples will be serialized, but not 100 lambdas + 100 tuples.

Reply:
Removed this method. Only ints are currently supported.